stdlib/Profile/src/Profile.jl

```diff
@@ -386,59 +485,97 @@ function fetch()
     end
     data = Vector{UInt}(undef, len)
     GC.@preserve data unsafe_copyto!(pointer(data), get_data_pointer(), len)
-    return data
+    if include_meta
+        return data
```

Comment:
I remember we originally discussed one possible approach to this would be to insert the information about the metadata into the `lidict` returned by `retrieve()`, so that the metadata frames can be treated as "normal frames" by the downstream consumers?

Something like teaching `getdict` to construct a `Base.StackTraces.StackFrame` for those metadata frames.

------

Does anyone ever consume the data from `fetch()` directly, by looking up the IP addresses themselves?

Reply:
In here somewhere perhaps:
https://github.com/JuliaLang/julia/blob/247b65eadf1271cac35e86998537a94ed15304b4/stdlib/Profile/src/Profile.jl#L299-L312

Reply:
> Does anyone ever consume the data from fetch() directly, by looking up the IP addresses themselves?

We do xD https://github.com/JuliaPerf/PProf.jl/blob/04b3157eb20b167239bc1a1ad9a7f95063e9eafa/src/PProf.jl#L160

> so that the metadata frames can be treated as "normal frames" by the downstream consumers?

I don't think that's possible, we have 3 frames now that are metadata and they would need to be collapsed to one, and would introduce ambiguity in the downstream consumer.

Reply:
Yeah, that's what I thought